packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -630,7 +630,20 @@ public boolean onActivityResult(
 
   private void handleChooseImageResult(int resultCode, Intent data) {
     if (resultCode == Activity.RESULT_OK && data != null) {
-      String path = fileUtils.getPathFromUri(activity, data.getData());
+      Uri uri = null;
+      // On several pre-Android 13 devices using Android Photo Picker, the Uri from getData() could be null.
+      if (data.getData() != null) {
+        uri = data.getData();
```

Comment:
Why is this being called twice, instead of:

```java
      Uri uri = data.getData();
      // On several pre-Android 13 devices using Android Photo Picker, the Uri from getData() could be null.
      if (uri == null && data.getClipData() != null && data.getClipData().getItemCount() == 1) {
        ...
```